src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const properties = [];
+
+  actions.forEach(object => {
+    switch (object.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, object.extraData);
+        properties.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        object.keysToRemove.forEach(key => {
+          delete stateCopy[key];
+        });
+        properties.push({ ...stateCopy });
```

Comment:
You are doing it 3 times
Move it out of the switch block - do it below only 1 time